include/merlin_hashtable.cuh

```diff
@@ -98,6 +98,7 @@ struct HashTableOptions {
   int device_id = -1;              ///< The ID of device.
   bool io_by_cpu = false;  ///< The flag indicating if the CPU handles IO.
   bool use_constant_memory = false;  ///< reserved
+  int reserved_key_index = 0; ///< The binary index of reserved key.
```

Comment:
Need more detail comment which allows user know the configuration effect exactly